core/src/main/java/org/apache/carbondata/core/metadata/schema/table/column/ColumnSchema.java

```diff
@@ -578,6 +581,7 @@ public void readFields(DataInput in) throws IOException {
       }
     }
     this.isLocalDictColumn = in.readBoolean();
+    this.indexColumn = in.readBoolean();
```

Comment:
Have you tested compatibility ? If the old version doesn't have this field. in.readBoolean() may fail

Reply:
I checked, it is for seriallization. Should not affect compatibility